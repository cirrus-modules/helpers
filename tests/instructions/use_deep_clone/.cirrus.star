load("cirrus", environ = "env")
load("../../../lib.star", "script", "task", "use_deep_clone")

def main():
    return [
        use_deep_clone(windows_task()),
        use_deep_clone(linux_task(), before = "test_script"),
    ]

def windows_task():
    return task(
        name = "Windows (windowsservercore 2019)",
        instance = {
            "windows_container": {
                "image": "python:3.8-windowsservercore",
                "os_version": 2019,
            },
        },
        instructions = [
            script("test", "python3 -c 'import sys; print(sys.version)'"),
        ],
    )

def linux_task():
    return task(
        name = "Linux (debian buster)",
        instance = {
            "container": {"image": "python:3.8-buster"},
        },
        instructions = [
            script("prepare", "python3 -m pip install -U pip setuptools"),
            script("test", "python3 -c 'import sys; print(sys.version)'"),
        ],
    )
