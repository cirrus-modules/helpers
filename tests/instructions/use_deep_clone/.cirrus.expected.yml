task:
  name: Windows (windowsservercore 2019)
  windows_container:
    image: python:3.8-windowsservercore
    os_version: 2019
  clone_script:
    - git clone --recursive https://x-access-token:%CIRRUS_REPO_CLONE_TOKEN%@github.com/%CIRRUS_REPO_FULL_NAME%.git %CIRRUS_WORKING_DIR%
    - git fetch origin pull/%CIRRUS_PR%/head:pull/%CIRRUS_PR%
    - git reset --hard %CIRRUS_CHANGE_IN_REPO%
  test_script:
    - python3 -c 'import sys; print(sys.version)'
task:
  name: Linux (debian buster)
  container:
    image: python:3.8-buster
  prepare_script:
    - python3 -m pip install -U pip setuptools
  clone_script:
    - git clone --recursive https://x-access-token:${CIRRUS_REPO_CLONE_TOKEN}@github.com/${CIRRUS_REPO_FULL_NAME}.git ${CIRRUS_WORKING_DIR}
    - git fetch origin pull/${CIRRUS_PR}/head:pull/${CIRRUS_PR}
    - git reset --hard ${CIRRUS_CHANGE_IN_REPO}
  test_script:
    - python3 -c 'import sys; print(sys.version)'
